### PR TITLE
flatten nested lists for flatten in visitor

### PR DESCRIPTION
**Pull request by deads2k**
Lists which contain lists should themselves be flattened by a flatten call to the resource builder

@kubernetes/sig-cli-maintainers 
@soltysh @juanvallejo 

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/genericclioptions/resource/visitor.go [hunk 1]
```diff
@@ -376,12 +376,28 @@ func (v FlattenListVisitor) Visit(fn VisitorFunc) error {
...
-		items, err := meta.ExtractList(info.Object)
-		if err != nil {
+		if !meta.IsListType(info.Object) {
 			return fn(info, nil)
 		}
-		if errs := runtime.DecodeList(items, v.mapper.decoder); len(errs) > 0 {
-			return utilerrors.NewAggregate(errs)
+
+		items := []runtime.Object{}
+		itemsToProcess := []runtime.Object{info.Object}
```
Comment (juanvallejo) [inline pkg/kubectl/genericclioptions/resource/visitor.go]: the length of this does not seem to change within the for loop below. Is a slice needed here?

Comment (liggitt) [reply pkg/kubectl/genericclioptions/resource/visitor.go]: I think line 400 is supposed to be appending to `itemsToProcess`

Comment (liggitt) [reply pkg/kubectl/genericclioptions/resource/visitor.go]: probably worth a test

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by deads2k** [comment]
/hold

I owe a test on this.  I'll get to it today.

**Comment by deads2k** [comment]
/hold cancel

fixed

File: pkg/kubectl/genericclioptions/resource/visitor.go (hunk 1 above)
Comment (deads2k) [reply pkg/kubectl/genericclioptions/resource/visitor.go]: > worth a test

added




> I think line 400 is supposed to be appending to itemsToProcess

fixed

**Pull request merged**